### PR TITLE
first draft a4 group and aggregate

### PR DESCRIPTION
**Pull request by jakemulf**

### CODE REVIEW

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Grouping and aggregating time series data
+
+In our previous notebook, we showed how to do time series and relational joins on our data. In this notebook, we show how to perform groupings and aggregations with our time series data.
+
+Let's start again by simulating some real-time data. This time we'll use a shorter time interval.
+
+```python
+time_interval = expressionToNanos("T1S")
```
Comment (hythloda) [inline demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:8]: should you have the imports so this works encase they don't do the prior one?

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+boolean_count_agg = as_list([
+    agg.AggCount("BooleanCount")
+])
+boolean_counts = daily_data.aggBy(boolean_count_agg, "Boolean")
+
+character_count_agg = as_list([
+    agg.AggCount("CharacterCount")
+])
+character_counts = daily_data.aggBy(character_count_agg, "Character")
+```
```
Comment (hythloda) [inline demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:52]: should have some combined aggregations?

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md (hunk 1 above)
Comment (jakemulf) [reply demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:8]: Yeah I thought about that and I think you're right. I'll change this (and the other notebooks) to be more verbose

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md (hunk 2 above)
Comment (jakemulf) [reply demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:52]: Yeah I can add another one that's a combined aggregation

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Grouping and aggregating time series data
+
+In our previous notebook, we showed how to do time series and relational joins on our data. In this notebook, we show how to perform groupings and aggregations with our time series data.
+
+Let's start again by simulating some real-time data. This time we'll use a shorter time interval.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:5]:
```suggestion
Let's start again by simulating some real-time data. Here, we'll use a shorter time interval.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+Let's group this data a few different ways based on the columns we have.
+
+```python
+group_by_character = daily_data.groupBy("Character")
+group_by_boolean = daily_data.groupBy("Boolean")
+group_by_odd_even = daily_data.update("IsEven = (Number % 2 == 0)").groupBy("IsEven").dropColumns("IsEven")
+```
+
+Just like all other real-time operations in Deephaven, these groupings update in real-time as the source table updates.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:25]:
```suggestion
Just like all other real-time operations in Deephaven, these groupings update in real time as the source table updates.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+Aggregations allow you to perform operations on grouped data. Let's perform some aggregations to compute statistics on our data.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:27]:
```suggestion
Aggregations allow you to perform operations on grouped data. The query below computes statistics on our data.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+character_counts = daily_data.aggBy(character_count_agg, "Character")
+
+combo_count_avg_agg = as_list([
+    agg.AggCount("ComboCount"),
+    agg.AggAvg("AvgNumber = Number")
+])
+combo_counts_avg = daily_data.aggBy(combo_count_avg_agg, "Boolean", "Character")
+```
+
+Just like the grouped data, these aggregations update in real-time. You don't need to recompute any of the averages or counts, Deephaven natively supports these real-time updates.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A4 Group and aggregate .md:62]:
```suggestion
Just like the grouped data, these aggregations update in real time. You don't need to recompute any of the averages or counts, Deephaven natively supports these real-time updates.
```

**Review by margaretkennedy — APPROVED** [review]

**Review by margaretkennedy — APPROVED** [review]

**Pull request merged**